js/src/admin/components/AuthPage.js

```diff
@@ -0,0 +1,207 @@
+import Page from './Page';
+import Checkbox from '../../common/components/Checkbox';
+import Button from '../../common/components/Button';
+import Alert from '../../common/components/Alert';
+import Switch from '../../common/components/Switch';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+import ItemList from '../../common/utils/ItemList';
+import icon from '../../common/helpers/icon';
+import saveSettings from '../utils/saveSettings';
+
+export default class AuthPage extends Page {
+  init() {
+    super.init();
+
+    this.saving = false;
+    this.refresh();
+  }
+
+  refresh() {
+    this.loading = true;
+
+    this.drivers = app.data.ssoDrivers;
+    this.driverFields = this.driverFieldsList().toArray();
+    this.driverInputs = [];
+
+    this.fields = ['allow_sign_up', 'enable_user_pass_auth'];
+    this.values = {};
+
+    const settings = app.data.settings;
+    this.fields.forEach(key => this.values[key] = m.prop(settings[key]));
+
+    for (const driver in this.drivers) {
+      this.driverFields.forEach(field => {
+        const fieldName = this.driverFieldKey(field.name, driver);
+
+        this.fields.push(fieldName);
+        this.values[fieldName] = m.prop(settings[fieldName]);
+
+        this.driverInputs[fieldName] = new Checkbox({
```

Comment:
Component instances should be avoided as much as possible. Try using `Checkbox.component()` in the view where it's needed. This syntax will likely no longer work once we update to the newer Mithril.

Reply:
Fixed, although when I tried to do this with the Alert, it didn't work so I didn't include that. Is that something we've encountered before?

Reply:
`Alert` is different. An instance has to be passed to `app.alerts.show` so while it also inherits from Component, it's not used in the same way as components injected in a `view` method.